src/send/SendSelectRecipient.tsx

```diff
@@ -130,32 +127,164 @@ const getStartedStyles = StyleSheet.create({
+  recipient,
+  recipientVerificationStatus,
+  onPress,
+}: {
+  recipient: Recipient | null
+  recipientVerificationStatus: RecipientVerificationStatus
+  onPress: (shouldInviteRecipient: boolean) => void
+}) {
   const { t } = useTranslation()
+  const sendOrInviteButtonDisabled =
+    !!recipient && recipientVerificationStatus === RecipientVerificationStatus.UNKNOWN
+  const shouldInviteRecipient =
+    !sendOrInviteButtonDisabled &&
+    !!recipient?.e164PhoneNumber &&
+    recipientVerificationStatus === RecipientVerificationStatus.UNVERIFIED
+  return (
+    <Button
+      testID="SendOrInviteButton"
+      style={styles.sendOrInviteButton}
+      onPress={() => onPress(shouldInviteRecipient)}
+      disabled={sendOrInviteButtonDisabled}
+      text={shouldInviteRecipient ? t('invite') : t('send')}
+      size={BtnSizes.FULL}
+    />
+  )
+}
+enum SelectRecipientView {
+  Recent = 'Recent',
+  Contacts = 'Contacts',
+}
+
+function SendSelectRecipient({ route }: Props) {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
 
-  const [searchQuery, setSearchQuery] = useState('')
-  const throttledSearch = throttle((searchInput: string) => {
-    setSearchQuery(searchInput)
-  }, SEARCH_THROTTLE_TIME)
+  const forceTokenId = route.params?.forceTokenId
+  const defaultTokenIdOverride = route.params?.defaultTokenIdOverride
 
-  const [showContacts, setShowContacts] = useState(false)
+  const [showSendOrInviteButton, setShowSendOrInviteButton] = useState(false)
 
-  const recentRecipients = useSelector((state) => state.send.recentRecipients)
-  const contactsCache = useSelector(phoneRecipientCacheSelector)
-  const contactRecipients = useMemo(
-    () => sortRecipients(Object.values(contactsCache)),
-    [contactsCache]
-  )
-  const showGetStarted = !recentRecipients.length
+  const [showSearchResults, setShowSearchResults] = useState(false)
+
+  const [activeView, setActiveView] = useState(SelectRecipientView.Recent)
 
-  const { recipientVerificationStatus, recipient, setSelectedRecipient } =
+  const [showInviteModal, setShowInviteModal] = useState(false)
+
+  const onSearch = (searchQuery: string) => {
+    // Always unset the selected recipient and hide the send/invite button
+    // when the search query is changed in order to prevent edge cases
+    // where the button appears but is bound to a recipient that is
+    // not present on the page.
+    unsetSelectedRecipient()
+    setShowSendOrInviteButton(false)
+    setShowSearchResults(!!searchQuery)
+  }
+  const { contactRecipients, recentRecipients } = useSendRecipients()
+  const { mergedRecipients, searchQuery, setSearchQuery } = useMergedSearchRecipients(onSearch)
+
+  const { recipientVerificationStatus, recipient, setSelectedRecipient, unsetSelectedRecipient } =
     useFetchRecipientVerificationStatus()
 
-  const dispatch = useDispatch()
+  const setSelectedRecipientWrapper = (selectedRecipient: Recipient) => {
+    setSelectedRecipient(selectedRecipient)
```

Comment:
should we include an analytics event here? We don't show the send button for recent recipients right?

Reply:
Ah yeah, good catch